### PR TITLE
fix(parse_query_string): only the first set of values for the same key are are split

### PR DESCRIPTION
**Pull request by kgriffs**
Address #1777

To Do:

- [x] Decide if this is a good idea. 😄 
- [x] Add news fragment(s)
- [x] Port fix to pyx implementation
- [x] Ensure full test coverage

### CODE REVIEW

**Comment by kgriffs** [comment]
@vytas7 if this PR seems like a good idea, do you want to contribute the cython fix?

**Comment by vytas7** [comment]
Sure. It should be fairly simple since that code path is not fully cythonized.

And I do think it is a good idea, I'm not sure I understood the original intent tbh. It could have been an optimization, but the case itself seems to be quite uncommon: normally you would choose either or.

**Comment by vytas7** [comment]
@kgriffs Done!
While at it, I've also removed one roundtrip via Python call, but I haven't benchmarked much. It _should_ be a slight improvement, although the whole csv path isn't particularly optimized, it's rather cythonized Python code.

**Comment by kgriffs** [comment]
>  but the case itself seems to be quite uncommon: normally you would choose either or.

Agreed. However we have a reputation to uphold. 😸

File: falcon/request.py [hunk 1]
```diff
@@ -1937,13 +1937,16 @@ class RequestOptions:
...
+            When ``False``,
             values containing commas are left as-is. In this mode, list items
             are taken only from multiples of the same parameter name within the
-            query string (i.e. ``/?t=1,2,3&t=4`` becomes ``['1,2,3', '4']``).
+            query string (i.e. ``t=1,2,3&t=4`` becomes ``['1,2,3', '4']``).
+
             When `auto_parse_qs_csv` is set to ``True``, the query string value
             is also split on non-percent-encoded commas and these items
-            are added to the final list (i.e. ``/?t=1,2,3&t=4``
+            are added to the final list (i.e. ``t=1,2,3&t=4``
```
Comment (CaselIT) [inline falcon/request.py:1949]: we could use as example `t=1,2,3&t=4,5` so it's clear what the behaviour is

File: tests/test_query_params.py [hunk 2]
```diff
@@ -182,6 +182,67 @@ def test_option_auto_parse_qs_csv_simple_true(self, simulate_request, client, re
...
+        query_string = qs
+        simulate_request(client=client, path='/', query_string=query_string)
+
+        req = resource.captured_req
+
+        # Old Behavior
+        #
+        # assert req.params['t'] == ['1', '2', '3,4']
+        # assert req.get_param('t') in ['1', '2', '3,4']
+        # assert req.get_param_as_list('t') == ['1', '2', '3,4']
```
Comment (CaselIT) [inline tests/test_query_params.py:239]: I think this can be cleaned up

**Review by CaselIT — CHANGES_REQUESTED** [review]
Looks ok, just a documentation improvement

File: tests/test_query_params.py (hunk 2 above)
Comment (kgriffs) [reply tests/test_query_params.py:239]: Good catch. I had that in there for purposes of discussion.

**Comment by kgriffs** [comment]
@vytas7 thanks for your help on this!

**Review by CaselIT — APPROVED** [review]

**Review by vytas7 — APPROVED** [review]
LGTM 👍 

(I hope that piece of Cython was carefully reviewed by someone else than me, too 🙂)

**Comment by CaselIT** [comment]
> LGTM 👍 
> 
> (I hope that piece of Cython was carefully reviewed by someone else than me, too 🙂)  

It looked ok to me

**Pull request merged**